EIPS/eip-3076.md

```diff
@@ -0,0 +1,292 @@
+
+2. Refuse to sign any block with `slot <= min(b.slot for b in data.signed_blocks if b.pubkey == proposer_pubkey)`, except if it is a repeat signing as determined by the `signing_root`.
+
+3. Refuse to sign any attestation that is slashable with respect to the attestations contained in `data.signed_attestations`. For details of what constitutes a slashable attestation, see [is_slashable_attestation_data][isad].
+4. Refuse to sign any attestation with source epoch less than the minimum source epoch present in that signer's attestations (as seen in `data.signed_attestations`). In pseudocode:
+```python3
+source.epoch <
+    min(att.source_epoch
+        for att in data.signed_attestations
+        if att.pubkey == attester_pubkey)
+```
+
+5. Refuse to sign any attestation with target epoch less than or equal to the minimum target epoch present in that signer's attestations (as seen in `data.signed_attestations`). In pseudocode:
+
+```python3
+target_epoch <=
+    min(att.target_epoch
+        for att in data.signed_attestations
+        if att.pubkey == attester_pubkey)
+```
+
+### Additional Information
+
+- The `interchange_format_version` version is set to 5.
+
+- A signed block or attestation's `signing_root` refers to the message data (hash tree root) that gets signed with a BLS signature. It  allows validators to re-sign and re-broadcast blocks or attestations if asked.
+
+- The `signed_blocks` `signing_root`s are calculated using [`compute_signing_root(block, domain)`][csr]: where `block` is the block (of type `BeaconBlock` or `BeaconBlockHeader`) that was signed, and `domain` is equal to [`compute_domain(DOMAIN_BEACON_PROPOSER, fork, metadata.genesis_validators_root)`][cd].
+
+- The `signed_attestations` `signing_root`s are calculated using [`compute_signing_root(attestation, domain)`][csr]: where `attestation` is the attestation (of type `AttestationData`) that was signed, and `domain` is equal to [`compute_domain(DOMAIN_BEACON_ATTESTER, fork, metadata.genesis_validators_root)`][cd].
+
+[pps]: https://github.com/ethereum/eth2.0-specs/blob/v1.0.0-rc.0/specs/phase0/beacon-chain.md#proposer-slashings
+[isad]: https://github.com/ethereum/eth2.0-specs/blob/v1.0.0-rc.0/specs/phase0/beacon-chain.md#is_slashable_attestation_data
+[csr]: https://github.com/ethereum/eth2.0-specs/blob/v1.0.0-rc.0/specs/phase0/beacon-chain.md#compute_signing_root
+[cd]: https://github.com/ethereum/eth2.0-specs/blob/v1.0.0-rc.0/specs/phase0/beacon-chain.md#compute_domain
+
+
+## Rationale
+
+The specification is designed to be flexible enough to support the full variety of slashing protection strategies that clients may implement.
+
+An earlier design encompassed two separate formats to achieve the same goal: `complete` and `minimal` (for the complete changelog [see here](https://hackmd.io/@sproul/Bk0Y0qdGD)).
```

Comment:
It is best to just explain the rationale for a decision and leave it at that.  Linking out to previous implementations will almost certainly result in dead links and for future readers it can be confusing.  In this case, something like this seems like it would be fine:
```suggestion
We considered using two separate formats (`complete` and `minimal`) rather than a single consolidated format but found that ran into problems.
```